src/libraries/System.Text.Json/gen/JsonSourceGenerator.Emitter.cs

```diff
@@ -728,7 +730,16 @@ private static string GeneratePropMetadataInitFunc(TypeGenerationSpec typeGenera
         JsonPropertyName = {jsonPropertyNameValue}
     }};
 
-    {PropVarName}[{i}] = {JsonMetadataServicesTypeRef}.CreatePropertyInfo<{memberTypeCompilableName}>({OptionsLocalVariableName}, {infoVarName});
+    {JsonPropertyInfoTypeRef} {propertyInfoVarName} = {JsonMetadataServicesTypeRef}.CreatePropertyInfo<{memberTypeCompilableName}>({OptionsLocalVariableName}, {infoVarName});");
+
+                    if (memberMetadata.IsRequired)
+                    {
+                        sb.Append($@"
```

Comment:
Shouldn't this be `AppendLine`? What does generated code look like here?

Reply:
> What does generated code look like here?

That's going to depend on whether the generator itself is compiled on Windows or Unix.  We'd like to avoid such differences.

Reply:
I followed the existing patterns here. If we want to make any changes probably it would make sense to make them globally. As for generated code, here is example with required and non-required property:

```csharp
            private static global::System.Text.Json.Serialization.Metadata.JsonPropertyInfo[] ClassWithRequiredCustomAttributesPropInit(global::System.Text.Json.JsonSerializerOptions options)
            {
                global::System.Text.Json.Serialization.Metadata.JsonPropertyInfo[] properties = new global::System.Text.Json.Serialization.Metadata.JsonPropertyInfo[3];
            
                global::System.Text.Json.Serialization.Metadata.JsonPropertyInfoValues<global::System.String> info0 = new global::System.Text.Json.Serialization.Metadata.JsonPropertyInfoValues<global::System.String>()
                {
                    IsProperty = true,
                    IsPublic = true,
                    IsVirtual = false,
                    DeclaringType = typeof(global::System.Text.Json.Serialization.Tests.DefaultJsonPropertyInfoTests.ClassWithRequiredCustomAttributes),
                    Converter = null,
                    Getter = static (obj) => ((global::System.Text.Json.Serialization.Tests.DefaultJsonPropertyInfoTests.ClassWithRequiredCustomAttributes)obj).NonRequired!,
                    Setter = static (obj, value) => ((global::System.Text.Json.Serialization.Tests.DefaultJsonPropertyInfoTests.ClassWithRequiredCustomAttributes)obj).NonRequired = value!,
                    IgnoreCondition = null,
                    HasJsonInclude = false,
                    IsExtensionData = false,
                    NumberHandling = default,
                    PropertyName = "NonRequired",
                    JsonPropertyName = null
                };
            
                global::System.Text.Json.Serialization.Metadata.JsonPropertyInfo propertyInfo0 = global::System.Text.Json.Serialization.Metadata.JsonMetadataServices.CreatePropertyInfo<global::System.String>(options, info0);
                properties[0] = propertyInfo0;
            
                global::System.Text.Json.Serialization.Metadata.JsonPropertyInfoValues<global::System.String> info1 = new global::System.Text.Json.Serialization.Metadata.JsonPropertyInfoValues<global::System.String>()
                {
                    IsProperty = true,
                    IsPublic = true,
                    IsVirtual = false,
                    DeclaringType = typeof(global::System.Text.Json.Serialization.Tests.DefaultJsonPropertyInfoTests.ClassWithRequiredCustomAttributes),
                    Converter = null,
                    Getter = static (obj) => ((global::System.Text.Json.Serialization.Tests.DefaultJsonPropertyInfoTests.ClassWithRequiredCustomAttributes)obj).RequiredA!,
                    Setter = static (obj, value) => ((global::System.Text.Json.Serialization.Tests.DefaultJsonPropertyInfoTests.ClassWithRequiredCustomAttributes)obj).RequiredA = value!,
                    IgnoreCondition = null,
                    HasJsonInclude = false,
                    IsExtensionData = false,
                    NumberHandling = default,
                    PropertyName = "RequiredA",
                    JsonPropertyName = null
                };
            
                global::System.Text.Json.Serialization.Metadata.JsonPropertyInfo propertyInfo1 = global::System.Text.Json.Serialization.Metadata.JsonMetadataServices.CreatePropertyInfo<global::System.String>(options, info1);
                propertyInfo1.IsRequired = true;
                properties[1] = propertyInfo1;
            
                global::System.Text.Json.Serialization.Metadata.JsonPropertyInfoValues<global::System.String> info2 = new global::System.Text.Json.Serialization.Metadata.JsonPropertyInfoValues<global::System.String>()
                {
                    IsProperty = true,
                    IsPublic = true,
                    IsVirtual = false,
                    DeclaringType = typeof(global::System.Text.Json.Serialization.Tests.DefaultJsonPropertyInfoTests.ClassWithRequiredCustomAttributes),
                    Converter = null,
                    Getter = static (obj) => ((global::System.Text.Json.Serialization.Tests.DefaultJsonPropertyInfoTests.ClassWithRequiredCustomAttributes)obj).RequiredB!,
                    Setter = static (obj, value) => ((global::System.Text.Json.Serialization.Tests.DefaultJsonPropertyInfoTests.ClassWithRequiredCustomAttributes)obj).RequiredB = value!,
                    IgnoreCondition = null,
                    HasJsonInclude = false,
                    IsExtensionData = false,
                    NumberHandling = default,
                    PropertyName = "RequiredB",
                    JsonPropertyName = null
                };
            
                global::System.Text.Json.Serialization.Metadata.JsonPropertyInfo propertyInfo2 = global::System.Text.Json.Serialization.Metadata.JsonMetadataServices.CreatePropertyInfo<global::System.String>(options, info2);
                propertyInfo2.IsRequired = true;
                properties[2] = propertyInfo2;
            
                return properties;
            }
```

Reply:
I hadn't realized that formatting relies on line breaks in the string literal. Given that that's what we do in general, I guess it's fine.